app/controllers/supplementary/DeclarationAdditionalActorsController.scala

```diff
@@ -34,31 +43,123 @@ class DeclarationAdditionalActorsController @Inject()(
   appConfig: AppConfig,
   override val messagesApi: MessagesApi,
   authenticate: AuthAction,
+  errorHandler: ErrorHandler,
   customsCacheService: CustomsCacheService
 )(implicit ec: ExecutionContext)
     extends FrontendController with I18nSupport {
 
-  import DeclarationAdditionalActors._
-
-  val additionalActorsForm = form()
+  val exceedMaximumNumberError = "supplementary.additionalActors.maximumAmount.error"
+  val duplicateActorError = "supplementary.additionalActors.duplicated.error"
 
   def displayForm(): Action[AnyContent] = authenticate.async { implicit request =>
-    customsCacheService.fetchAndGetEntry[DeclarationAdditionalActors](supplementaryCacheId, formId).map {
-      case Some(data) => Ok(declaration_additional_actors(appConfig, additionalActorsForm.fill(data)))
-      case _          => Ok(declaration_additional_actors(appConfig, additionalActorsForm))
+    customsCacheService.fetchAndGetEntry[DeclarationAdditionalActorsData](supplementaryCacheId, formId).map {
+      case Some(data) => Ok(declaration_additional_actors(appConfig, form(), data.actors))
+      case _          => Ok(declaration_additional_actors(appConfig, form(), Seq()))
     }
   }
 
-  def saveAdditionalActors(): Action[AnyContent] = authenticate.async { implicit request =>
-    form
-      .bindFromRequest()
-      .fold(
-        (formWithErrors: Form[DeclarationAdditionalActors]) =>
-          Future.successful(BadRequest(declaration_additional_actors(appConfig, formWithErrors))),
-        form =>
-          customsCacheService.cache[DeclarationAdditionalActors](supplementaryCacheId, formId, form).map { _ =>
-            Redirect(controllers.supplementary.routes.DeclarationHolderController.displayForm())
-        }
-      )
+  def saveAndContinue(
```

Comment:
Please, change the order of methods, so that the caller is before callee.
And some of methods does not have to be public.

Reply:
👍